### PR TITLE
flag to copy existing capabilities for new role

### PR DESCRIPTION
**Pull request by shawnhooper**
When a new role is being created, allow capabilities to be copied from
an existing role. First attempt at resolving issue #393. 

Open to feedback on this approach.

### CODE REVIEW

**Comment by danielbachhuber** [comment]
Seems reasonable. CI is choking because your indentation is off.

1. What do you think about renaming the parameter to `--clone=<role>` ?
1. Can you add [functional tests](https://github.com/wp-cli/wp-cli/blob/master/CONTRIBUTING.md#functional-tests)?

**Comment by shawnhooper** [comment]
I saw the CI failed. I'll fix that when get home. Wrote that patch while
travelling, Still haven't quite tweaked my laptop settings.

I like the clone name. Shorter and clear.

I'll take a stab at the tests. May ask for help once I get into it.

Cheers.

**Comment by danielbachhuber** [comment]
> I'll take a stab at the tests. May ask for help once I get into it.

Sounds good — happy to help!

**Comment by szepeviktor** [comment]
We have a new abstracted function
https://github.com/wp-cli/wp-cli/blob/master/php/utils.php#L524
Please use it.

**Comment by szepeviktor** [comment]
No! `$clone` here:
https://github.com/wp-cli/wp-cli/pull/1863/files#diff-6abfcce1c4d661e4499fd90870919302R132

File: php/commands/role.php [hunk 1]
```diff
@@ -92,25 +92,55 @@ public function exists( $args ) {
...
 
-		if ( ! add_role( $role_key, $role_name ) )
+		if ( ! add_role( $role_key, $role_name ) ) {
 			WP_CLI::error( "Role couldn't be created." );
+		}
+
 		else
 			WP_CLI::success( sprintf( "Role with key %s created.", $role_key ) );
+
+		if ( isset( $assoc_args['clone'] ) ) {
```
Comment (szepeviktor) [inline php/commands/role.php]: `$clone` here,

File: php/commands/role.php [hunk 2]
```diff
@@ -92,25 +92,55 @@ public function exists( $args ) {
...
 
 		if ( empty( $role_key ) || empty( $role_name ) )
 			WP_CLI::error( "Can't create role, insufficient information provided.");
 
-		if ( ! add_role( $role_key, $role_name ) )
+		if ( ! add_role( $role_key, $role_name ) ) {
 			WP_CLI::error( "Role couldn't be created." );
+		}
+
 		else
```
Comment (szepeviktor) [inline php/commands/role.php]: Also `else` needs {}.

File: php/commands/role.php [hunk 3]
```diff
@@ -92,25 +92,55 @@ public function exists( $args ) {
...
 			WP_CLI::success( sprintf( "Role with key %s created.", $role_key ) );
+
+		if ( isset( $assoc_args['clone'] ) ) {
+			$target_role_obj = get_role( $role_key );
+			$count = 0;
+
+			foreach ( $source_role_obj->capabilities as $cap=>$value ) {
```
Comment (szepeviktor) [inline php/commands/role.php]: Please consider `$cap => $value` for readability.

**Comment by szepeviktor** [comment]
Try writing adding tests to `role.feature`

- first user has no role, then clone
- first user has 1 role
- first user has several roles

**Comment by shawnhooper** [comment]
@szepeviktor I've never written tests in the .feature file format before.  Reading up on them, but not quite comfortable writing them yet.

**Comment by szepeviktor** [comment]
It's like simple English.
I usually copy from other tests and modify a bit. Need *no* coding.

**Comment by danielbachhuber** [comment]
@shawnhooper any luck with the tests?

**Comment by danielbachhuber** [comment]
Sorted this out in #1948

**Pull request closed without merging**